tools/unicode-check.py

```diff
@@ -10,8 +10,13 @@
 # The set of Unicode code points greater than 127 that we
 # allow in the source code.
 box_drawing_chars = set(chr(cp) for cp in range(0x2500, 0x2580))
-allowed = (set(['®', 'é', 'ê', 'ó', 'ö', 'ü', 'λ', 'π', 'ω', '∫', '≠', '≥']) |
-           box_drawing_chars)
+greek_letters = set(chr(cp) for cp in chain(
```

Comment:
> The comment here and the way the code is written makes it appear that it may have been a deliberate decision not to allow all Greek letters. Before making this change you should probably find out why all Greek letters are not currently allowed.

I don't really make the same conclusion. Nothing explicit there that suggests that greek letters shouldn't be included especially when there are already 3 in that set.

> It's up to the maintainers to make a decision on this one.

@WarrenWeckesser I think you were original author of this script which replaced an old travis check. This PR introduced comments that include alpha, beta and gamma greek letters. Rather than just add those 3 characters to the `alllowed` set I future proofed the script to allow all greek letters (mathematical documentation relies heavily on greek letters). Can you see an issue with this change?

Reply:
OK, I think you may be right. I checked the PR where the unicode check was introduced. @WarrenWeckesser's rationale in the comment below seems to support your understanding. That particular PR was being conservative in what was excepted, but it's mentioned in another comment that they planned to loosen the restrictions as needed.

https://github.com/scipy/scipy/pull/12393#issuecomment-653179900
> Thanks Eric, and everyone else who commented.
> 
> I think we should have no problem adding printable characters from the latin1 character set (e.g. ç, ï, ñ, etc) when they occur in proper names of authors referenced in docstrings. The main goal is to avoid non-printing characters or non-ASCII variations of dashes and other punctuation that tend to sneak into docstrings with text that was cut-and-pasted from other sources.